src/tool/GenerateAction.cpp

```diff
@@ -20,11 +21,43 @@
 #include <mrdocs/Support/Path.hpp>
 #include <clang/Tooling/JSONCompilationDatabase.h>
 
+
 #include <cstdlib>
 
 namespace clang {
 namespace mrdocs {
 
+Expected<std::string>
+generateCompilationDatabaseIfNeeded(llvm::StringRef projectPath)
```

Comment:
Just `generateCompilationDatabase` should be fine. The combination of functions that already return `Expected` with `maybe_do_X`, `do_X_if_needed`, `try_to_do_X`, is dangerously viral. If it's not needed, we can just return no result.

Just include some javadocs around the function in the header files and we're fine.